typing/tast_iterator.ml

```diff
@@ -0,0 +1,495 @@
+  | Tpat_array l -> List.iter (sub.pat sub) l
+  | Tpat_or (p1, p2, _) ->
+      sub.pat sub p1;
+      sub.pat sub p2
+  | Tpat_alias (p, _, _) -> sub.pat sub p
+  | Tpat_lazy p -> sub.pat sub p
+  | Tpat_exception p -> sub.pat sub p
+
+let expr sub {exp_extra; exp_desc; exp_env; _} =
+  let extra = function
+    | Texp_constraint cty -> sub.typ sub cty
+    | Texp_coerce (cty1, cty2) ->
+        opt (sub.typ sub) cty1;
+        sub.typ sub cty2
+    | Texp_newtype _ -> ()
+    | Texp_poly cto -> opt (sub.typ sub) cto
+  in
+  List.iter (fun (e, _, _) -> extra e) exp_extra;
+  sub.env sub exp_env;
+  match exp_desc with
+    | Texp_ident _  -> ()
+    | Texp_constant _ -> ()
+    | Texp_let (rec_flag, list, exp) ->
+        sub.value_bindings sub (rec_flag, list);
+        sub.expr sub exp
+    | Texp_function {cases; _} -> sub.cases sub cases
+    | Texp_apply (exp, list) ->
+          sub.expr sub exp;
+          List.iter (fun (_, o) -> opt (sub.expr sub) o) list
+    | Texp_match (exp, cases, _) ->
+          sub.expr sub exp;
+          sub.cases sub cases
+    | Texp_try (exp, cases) ->
+          sub.expr sub exp;
+          sub.cases sub cases
+    | Texp_tuple list -> List.iter (sub.expr sub) list
+    | Texp_construct (_, _, args) -> List.iter (sub.expr sub) args
+    | Texp_variant (_, expo) -> opt (sub.expr sub) expo
+    | Texp_record { fields; extended_expression; _} ->
+        Array.iter (function
+            | _, Kept _ -> ()
+            | _, Overridden (_, exp) -> sub.expr sub exp)
+            fields;
+        opt (sub.expr sub) extended_expression;
+    | Texp_field (exp, _, _) -> sub.expr sub exp
+    | Texp_setfield (exp1, _, _, exp2) ->
+        sub.expr sub exp1;
+        sub.expr sub exp2
+    | Texp_array list -> List.iter (sub.expr sub) list
+    | Texp_ifthenelse (exp1, exp2, expo) ->
+        sub.expr sub exp1;
+        sub.expr sub exp2;
+        opt (sub.expr sub) expo
+    | Texp_sequence (exp1, exp2) ->
+        sub.expr sub exp1;
+        sub.expr sub exp2
+    | Texp_while (exp1, exp2) ->
+        sub.expr sub exp1;
+        sub.expr sub exp2
+    | Texp_for (_, _, exp1, exp2, _, exp3) ->
+        sub.expr sub exp1;
+        sub.expr sub exp2;
+        sub.expr sub exp3
+    | Texp_send (exp, _, expo) ->
+        sub.expr sub exp;
+        opt (sub.expr sub) expo
+    | Texp_new _ -> ()
+    | Texp_instvar _ -> ()
+    | Texp_setinstvar (_, _, _, exp) ->sub.expr sub exp
+    | Texp_override (_, list) ->
+        List.iter (fun (_, _, e) -> sub.expr sub e) list
+    | Texp_letmodule (_, _, _, mexpr, exp) ->
+        sub.module_expr sub mexpr;
+        sub.expr sub exp
+    | Texp_letexception (cd, exp) ->
+        sub.extension_constructor sub cd;
+        sub.expr sub exp
+    | Texp_assert exp -> sub.expr sub exp
+    | Texp_lazy exp -> sub.expr sub exp
+    | Texp_object (cl, _) -> sub.class_structure sub cl
+    | Texp_pack mexpr -> sub.module_expr sub mexpr
+    | Texp_letop {let_ = l; ands; body; _} ->
```

Comment:
Could just use punning for `let_` as well, instead of rebinding.

Reply:
I had to rebind let_ because for some reason my editor was convinced that it was a let expression and not a variable name. I can remove if you'd like, but it might cause issues for other people too.